src/compilers/component.cr

```diff
@@ -174,13 +180,15 @@ module Mint
           function =
             node.functions.find(&.name.value.==(key))
 
-          # If the user defined the same function the code goes after it.
-          if function && value
-            function.keep_name = true
+          function.keep_name = true if function
 
+          # If the user defined the same function the code goes after it.
+          if function && value.any?
             compile function, js.statements(value)
-          elsif !value.empty?
+          elsif value.any?
```

Comment:
`empty?` is in general more performant than `any?`.

```suggestion
          elsif !value.empty?
```

Reply:
Didn't know that, probably this should be a rule in Ameba or it should be optimized to check for emptiness and return early.

For me `any?` is more readable.